scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -272,6 +274,35 @@
          (catch Exception e
            (log/error e "Mesos scheduler status update error")))))
 
+(defn handle-framework-message
+  "Processes a framework message from Mesos."
+  [conn framework-message]
+  (timers/time!
+    handle-framework-message-duration
+    (try
+      (let [db (db conn)
+            {:strs [exit-code progress-message progress-percent sandbox-directory task-id] :as message}
+            (-> (String. ^bytes framework-message "UTF-8") (json/read-str))
+            _ (log/debug "Received framework message:" {:task-id task-id, :message message})
+            _ (when (str/blank? task-id)
+                (throw (ex-info "task-id is empty in framework message" {:message message})))
+            instance (ffirst (q '[:find ?i
```

Comment:
You can use a look-up ref (`[:instance/task-id task-id]`) with an entity like `(d/entity db [:instance/task-id task-id])` which I think is cleaner the query. Alternatively (or also) consider making this a function.